src/main/java/FuncTest/TestFunc/utils.java

```diff
@@ -63,4 +63,69 @@ protected static HttpResponseMessage createWebhookResponseContent(String resultT
 				.header("Content-Type", "application/json; charset=UTF-8").header("Accept", "application/json").build();
 	}
 
+	//returns null is the parameter is missing
+	protected static String getUserParam(JSONObject queryResult, String paramName) {
+		JSONObject parameters = queryResult.getJSONObject("parameters");
+		return !parameters.has(paramName) ? null : parameters.getString(paramName);
+	}
+
+	
+	protected static String buildFilteringQuery(String facultyName, Integer lectureHours
+			, Integer tutorialHours, String dateRangeA[]) {
+		String q = "select name, ID from dbo.Courses where 1 = 1";
+		
+		if(isValidFaculty(facultyName)) q += " and faculty = " + quote(facultyName);
```

Comment:
Also i think you should consider adding a limit to the returned results.

Reply:
What would be an appropriate limit in your opinion?

Also, the concatenation was used since any subset of values counts as a valid filtering query. Meaning, I don't have a uniform structure that I'm putting parameters into. If it's absolutely critical I can work around to incorporating it. How strongly do you feel towards the issue?

Reply:
I set 15 as an initial limit

Reply:
About the use of prepared statements i think this can be fixed later and we shouldn't take more time to look for a workaround.